src/app/staticanalysis/modules/thisLocal.js

```diff
@@ -0,0 +1,30 @@
+var name = 'this on local'
+var desc = 'Invocation of local functions via this'
+var categories = require('./categories')
+var common = require('./staticAnalysisCommon')
+
+function thisLocal () {
+  this.warningNodes = []
+}
+
+thisLocal.prototype.visit = function (node) {
+  if (common.isThisLocalCall(node)) this.warningNodes.push(node)
+}
+
+thisLocal.prototype.report = function (compilationResults) {
+  this.warningNowNodes = []
+  return this.warningNodes.map(function (item, i) {
+    return {
+      warning: `use of "this" for local functions: never use this to call local functions, it only consumes more gas than normal local calls.`,
```

Comment:
Please capitalize the first word of a full sentence (I would even capitalize the first word of a sequence of words that is not a full sentence).

Perhaps use "functions in the same contract" instead of "local functions" for the second occurance.